fission/src/mirabuf/ScoringZoneSceneObject.ts

```diff
@@ -0,0 +1,276 @@
+import { Array_ThreeMatrix4, JoltMat44_ThreeMatrix4, ThreeQuaternion_JoltQuat, ThreeVector3_JoltVec3 } from "@/util/TypeConversions"
+import MirabufSceneObject, { RigidNodeAssociate } from "./MirabufSceneObject"
+import JOLT from "@/util/loading/JoltSyncLoader"
+import World from "@/systems/World"
+import Jolt from "@barclah/jolt-physics"
+import * as THREE from "three"
+import { OnContactAddedEvent, OnContactRemovedEvent } from "@/systems/physics/ContactEvents"
+import SceneObject from "@/systems/scene/SceneObject"
+import { ScoringZonePreferences } from "@/systems/preferences/PreferenceTypes"
+import SimulationSystem from "@/systems/simulation/SimulationSystem"
+import PreferencesSystem from "@/systems/preferences/PreferencesSystem"
+
+class ScoringZoneSceneObject extends SceneObject {
+    //Official FIRST hex
+    static redMaterial =  new THREE.MeshPhongMaterial({
+        color: 0xED1C24,
+        shininess: 0.0,
+        opacity: 0.7,
+        transparent: true,
+    })
+    static blueMaterial = new THREE.MeshPhongMaterial({
+        color: 0x0066B3,
+        shininess: 0.0,
+        opacity: 0.7,
+        transparent: true,
+    })  //0x0000ff
```

Comment:
I'm not exactly sure how we'd do this, but in the future we may want to integrate these colors into the theme editor. I'm not sure why, but someone may want to change the scoring zone colors to their team color or something.

Reply:
We could make it work. When we have more time, let's add them.